libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp

```diff
@@ -62,6 +62,22 @@ AP_BattMonitor_Analog::AP_BattMonitor_Analog(AP_BattMonitor &mon,
     AP_BattMonitor_Backend(mon, mon_state, params)
 {
     AP_Param::setup_object_defaults(this, var_info);
+
+    // no other good way of setting these defaults
+    if (mon_state.instance == 1) {
+ #ifdef HAL_BATT2_VOLT_PIN
+        _volt_pin.set_default(HAL_BATT2_VOLT_PIN);
```

Comment:
we better check that the boards that have these defines already are consistent with their defaults.parm, and remove from defaults.parm

Reply:
Done